python/lsst/daf/butler/registry/collections/_base.py

```diff
@@ -452,3 +453,94 @@ def _insert_collection_chain_rows(
             for child in child_keys
         ]
         self._db.insert(self._tables.collection_chain, *rows)
+
+    def prepend_collection_chain(
+        self, parent_collection_name: str, child_collection_names: list[str]
+    ) -> None:
```

Comment:
I think we want to insert some kind of cache-invalidation or cache check on `self._caching_context` here.  I think
```
if self._caching_context.is_enabled:
    raise <something>
```
would be fine, as I'm not sure we actually need to support modifying a collection while a caching context is active.  Same for `update_chain`.

Reply:
It does handle updating the cache at the end already.  But I like forbidding it entirely, that will let me remove a decent amount of code.

Reply:
It turns out that `setCollectionChain` is called inside a caching context from `Butler.import_`.  Whether that caching context is useful or not I can't really say.

So I'll do this in prepend_collection_chain but leave update_chain alone for now.